Simplified/Book/UI/NYPLBookCellDelegate.m

```diff
@@ -413,33 +425,50 @@ - (void)audiobookManagerDidRequestRefresh {
   [[NYPLBookRegistry sharedRegistry] setState:NYPLBookStateDownloadNeeded forIdentifier:self.book.identifier];
 
 #if FEATURE_OVERDRIVE
-  [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(updateODAudiobookManifest) name:NSNotification.NYPLMyBooksDownloadCenterDidChange object:nil];
+  [[NSNotificationCenter defaultCenter] addObserver:self
+                                           selector:@selector(updateODAudiobookManifest)
+                                               name:NSNotification.NYPLMyBooksDownloadCenterDidChange
+                                             object:nil];
 #endif
   [[NYPLMyBooksDownloadCenter sharedDownloadCenter] startDownloadForBook:self.book];
 }
 
 #if FEATURE_OVERDRIVE
 - (void)updateODAudiobookManifest {
-  if ([[NYPLBookRegistry sharedRegistry] stateForIdentifier:self.book.identifier] == NYPLBookStateDownloadSuccessful) {
+  NYPLBookState bookState = [[NYPLBookRegistry sharedRegistry]
+                             stateForIdentifier:self.book.identifier];
+
+  if (bookState == NYPLBookStateDownloadFailed) {
+    [[NSNotificationCenter defaultCenter] removeObserver:self];
+    [self.refreshAudiobookLock unlock];
+    return;
+  }
+
+  if (bookState == NYPLBookStateDownloadSuccessful) {
+    if (![(NSObject*)self.manager.audiobook isKindOfClass:[OverdriveAudiobook class]]) {
+      [[NSNotificationCenter defaultCenter] removeObserver:self];
+      [self.refreshAudiobookLock unlock];
+      return;
+    }
+
     OverdriveAudiobook *odAudiobook = (OverdriveAudiobook *)self.manager.audiobook;
 
     NSURL *const url = [[NYPLMyBooksDownloadCenter sharedDownloadCenter] fileURLForBookIndentifier:self.book.identifier];
     NSData *const data = [NSData dataWithContentsOfURL:url];
     if (data == nil) {
+      [[NSNotificationCenter defaultCenter] removeObserver:self];
```

Comment:
again here I don't know if we skipped the removal of the observer intentionally or if it is just a bug....

Reply:
Yeah I think we forgot to implement the removal of the observer, it should be fine to remove it before we unlock.

Reply:
well, we got a successful download and we tried to get the data for a given book, I think, but somehow we found no usable data. Since we are then presenting an error for a corrupted book (see next line) I suppose we are done and we should also unlock?